### ISSUE TITLE
Compile fails on newer mac SDKs (15.0+)

### ISSUE THREAD

**Issue by Proffles** [QSPFoundation/qgen#15]
Build fails on more recent Mac OSes. (15.0+).

error: 'CGDisplayCreateImage' is unavailable: obsoleted in macOS 15.0 - Please use ScreenCaptureKit instead.

See build fail snippet:

```
[ 54%] Building CXX object wxWidgets-build/libs/core/CMakeFiles/wxcore.dir/__/__/__/__/src/osx/carbon/dcscreen.cpp.o
/Users/davidbaker/development/qgen/build/wxWidgets-src/src/osx/carbon/dcscreen.cpp:72:13: error: 'CGDisplayCreateImage' is unavailable: obsoleted in macOS 15.0 - Please use ScreenCaptureKit instead.
   72 |     image = CGDisplayCreateImage(kCGDirectMainDisplay);
      |             ^
/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/System/Library/Frameworks/CoreGraphics.framework/Headers/CGDirectDisplay.h:383:33: note: 'CGDisplayCreateImage' has been explicitly marked unavailable here
  383 | CG_EXTERN CGImageRef __nullable CGDisplayCreateImage(CGDirectDisplayID displayID)
      |                                 ^
1 error generated.
make[2]: *** [wxWidgets-build/libs/core/CMakeFiles/wxcore.dir/__/__/__/__/src/osx/carbon/dcscreen.cpp.o] Error 1
make[1]: *** [wxWidgets-build/libs/core/CMakeFiles/wxcore.dir/all] Error 2
```

**Comment by hzdbyte** [comment]
It's caused by this: https://github.com/wxWidgets/wxWidgets/issues/24724
I'll try to update wxWidgets, mb they've fixed it already.

**Comment by hzdbyte** [comment]
Should be fixed by c4cd39e725d1eb11599220c52b16386fb3ee9a4c
